### ISSUE TITLE
520 RTSP error when trying to use TTS

### ISSUE THREAD

**Issue by smallgreyrock** [postlund/pyatv#1642]
### Describe the bug

I'm using the 3.3.1 beta with HomeAssistant 2022.2 and a Nabu Casa subscription (so using their TTS backend)

Individually, the cloud TTS and the appletv module both seem to work, but when I try to use the TTS feature of the appletv module to speak some text via a homepod or appletv, I get an error that doesn't mean anything obvious to me. The TV pairing process had three separate PINs, the Homepod pairing process didn't. I don't know if that's normal or not.

### Error log

```log


Logger: homeassistant.helpers.script.websocket_api_script
Source: custom_components/apple_tv/media_player.py:292
Integration: Apple TV
First occurred: February 2, 2022, 11:59:27 PM (2 occurrences)
Last logged: 12:03:55 AM
websocket_api script: Error executing script. Unexpected error for call_service at pos 1: RTSP/1.0 method SETUP failed with code 520: Origin Error

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 372, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 575, in _async_call_service_step
    await service_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1630, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1667, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/components/tts/__init__.py", line 205, in async_say_handle
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1630, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1667, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 204, in handle_service
    await self.hass.helpers.service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 668, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 921, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 705, in _handle_entity_call
    await result
  File "/config/custom_components/apple_tv/media_player.py", line 292, in async_play_media
    await self.atv.stream.stream_file(media_id)
  File "/usr/local/lib/python3.9/site-packages/pyatv/core/facade.py", line 382, in stream_file
    await self.relay("stream_file")(file, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/pyatv/protocols/raop/__init__.py", line 352, in stream_file
    await client.initialize(self.core.service.properties)
  File "/usr/local/lib/python3.9/site-packages/pyatv/protocols/raop/raop.py", line 457, in initialize
    await self._setup_session()
  File "/usr/local/lib/python3.9/site-packages/pyatv/protocols/raop/raop.py", line 486, in _setup_session
    resp = await self.rtsp.setup(
  File "/usr/local/lib/python3.9/site-packages/pyatv/support/rtsp.py", line 175, in setup
    return await self.exchange("SETUP", headers=headers, body=body)
  File "/usr/local/lib/python3.9/site-packages/pyatv/support/rtsp.py", line 258, in exchange
    resp = await self.connection.send_and_receive(
  File "/usr/local/lib/python3.9/site-packages/pyatv/support/http.py", line 412, in send_and_receive
    raise exceptions.HttpError(
pyatv.exceptions.HttpError: RTSP/1.0 method SETUP failed with code 520: Origin Error
```

### How to reproduce the bug?

Pick a supported homepod or appletv device, and attempt to use the Nabu Casa text to speech.

### What is expected behavior?

Hearing the text to speech voice played by the device.

### Operating System

Linux

### Python

3.9

### pyatv

0.10.0

### Device

Apple TV 4k 15.3

### Additional context

If there's any other info I can provide, please let me know.

**Comment by postlund** [comment]
The Apple TV requires three pins while the HomdPod does not require any, so that sounds OK to me. The "origin" error is a bit tricky. A similar error was reported about a year ago in the owntone project here:

https://github.com/owntone/owntone-server/issues/1154

I had some issues with it as well, but it kinda disappeared on its own at some point. Do you get the exact same error on both the Apple TV and HomePod? Do I understand you correctly that the cloud TTS works with some other media player you have? Maybe you can try playing a local file with `play_media` or pass for instance `https://file-examples-com.github.io/uploads/2017/11/file_example_MP3_1MG.mp3` and see if that works?

**Comment by smallgreyrock** [comment]
Cloud TTS plays in the browser, I don't have any other media players (aside from the tv/homepod) to test it against.

I don't think playing that mp3 works, I get `ERROR (MainThread) [custom_components.apple_tv.media_player] Media streaming is not possible with current configuration` from this:

```
service: media_player.play_media
data:
  media_content_id: https://file-examples-com.github.io/uploads/2017/11/file_example_MP3_1MG.mp3
  media_content_type: music
target:
  device_id: 82d46d029888feb4c043eaea6e463881
```

I might be doing it wrong of course

**Comment by tsroten** [comment]
Thanks for reporting this! I am also having the same issue. I first encountered this on pyatv 0.9.8 and hass-atv-beta 2.2.0, with Home Assistant 2021.12.9 (the first time I've used hass-atv-beta).

I'm now on pyatv 0.10.0, hass-atv-beta 3.0.1, and Home Assistant 2022.02.03, where I'm seeing the same problem. I've tested this using both Google Translate and Google Cloud Text-to-Speech with my HomePod Mini and Apple TV.

Both of those TTS platforms work on a different media player (Denon DRA-800H) using the Denon HEOS integration.

Using the `media_player.play_media` service, I can get that sample MP3 file to play on my Denon receiver, but I get the same RTSP 520 error when doing that on the HomePod Mini or Apple TV.

**Comment by tsroten** [comment]
I was able to fix this on my Raspberry Pi 4!

I successfully streamed to my HomePod Mini using `atvremote` on a different computer, which led me to look at my firewall configuration. Looking at my firewall logs on the RP4, I saw that the HomePod Mini was attempting to send UDP packets to ephemeral ports and they were being dropped. By opening up the ephemeral UDP port range (32768-65535) for traffic coming from the HomePod Mini, I was able to get the `SETUP` request to complete and the file to play successfully.
